### PR TITLE
feat(upgrade): print info links for Deno 2 RC releases

### PR DESCRIPTION
**Pull request by bartlomieju**
In addition to printing a blog post information (if it's available), this will also
print a link to migration guide and the bug tracker.

### CODE REVIEW

**Comment by bartlomieju** [comment]
Waiting on decision what's the blog post link before landing this one.

File: cli/tools/upgrade.rs [hunk 1]
```diff
@@ -230,11 +230,54 @@ fn get_minor_version(version: &str) -> &str {
...
+    && new_semver.minor == 0
+    && new_semver.patch == 0
+    && new_semver.pre.contains(&"rc".to_string());
+
+  if is_deno_2_rc {
+    log::info!(
+      "{}\n\n  {}\n",
+      colors::gray("Migration guide:"),
+      colors::bold(
+        "https://docs.deno.com/runtime/manual/advanced/migrate_deprecations/"
```
Comment (iuioiua) [inline cli/tools/upgrade.rs:255]:
```suggestion
        "https://docs.deno.com/runtime/manual/advanced/migrate_deprecations"
```

File: cli/tools/upgrade.rs [hunk 2]
```diff
@@ -226,15 +226,65 @@ impl<
...
+  let Ok(new_semver) = Version::parse_standard(new_version) else {
+    return;
+  };
+
+  let is_deno_2_rc = new_semver.major == 2
+    && new_semver.minor == 0
+    && new_semver.patch == 0
+    && new_semver.pre.first() == Some(&"rc".to_string());
+
+  if is_deno_2_rc {
```
Comment (lucacasonato) [inline cli/tools/upgrade.rs:257]: Probably in general when `new_semver.major == 2 && (current_semver.pre.first() == Some(&"rc".to_string() || current_semver.major == 1)`

That way 1.0 -> 2.0 (stable) upgrades will get this warning too.

Comment (bartlomieju) [reply cli/tools/upgrade.rs]: I updated it to that migration guide is printed for upgrading to Deno 2 RC, or when switching from Deno 1 to Deno 2.
I kept the remaining logic the same.

**Review by nathanwhit — APPROVED** [review]
LGTM

**Pull request merged**